### PR TITLE
added plugable interface - #623

### PR DESCRIPTION
**Pull request by krimdomu**
This is a pull request to allow the creation of new cli interfaces as discussed in #623 
This code tries to detect with `PPI` which cli interface it should load (by the help of a feature flag `cli_XYZ`)

So with this code it would be possible to experiment with new command line interfaces without compromising the current code.

### CODE REVIEW

File: bin/rex [hunk 1]
```diff
@@ -11,7 +11,14 @@ use warnings;
 
 # VERSION
 
-use Rex::CLI;
+use Rex::DetectCLI;
+my $cli = detect_cli;
+
+eval "require 'Rex/CLI/$cli.pm'";
```
Comment (kablamo) [inline bin/rex]: https://metacpan.org/pod/Module::Load?   Just a suggestion

File: lib/Rex/DetectCLI.pm [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+use vars qw(@EXPORT);
+
+use List::MoreUtils qw'firstidx';
+use Data::Dumper;
+
+use PPI;
+
+@EXPORT = qw(detect_cli);
+
+sub detect_cli {
```
Comment (kablamo) [inline lib/Rex/DetectCLI.pm]: This is pretty cool.  I mean its also insane.  I don't love it.  But I don't have any better ideas atm.  I thought PPI would be really slow.  Its not as bad as I thought though...

Comment (krimdomu) [inline lib/Rex/DetectCLI.pm]: yes, i know, i also feel not that good with it but it seems to work pretty good.

File: bin/rex (hunk 1 above)
Comment (krimdomu) [inline bin/rex]: you're right. and there is also: https://github.com/RexOps/Rex/blob/master/lib/Rex/Require.pm which is doing the same.

**Comment by krimdomu** [comment]
This can be closed in favor of Rex 2.0 which will implement a new cli

**Pull request closed without merging**